### ISSUE TITLE
[bug] STUCK in 'Accept connection request in the wallet'

### ISSUE THREAD

**Issue by fukemy** [reown-com/appkit#2750]
### Link to minimal reproducible example

https://stackblitz.com/edit/web3modal-next-template-zdukvm

### Summary

 I tested with the AppKit ethers with NextJS, follow the step from https://docs.walletconnect.com/appkit/next/core/installation
 When choose TrustWallet or MetaMask, then it stuck in 'Accept connection request in the wallet'
 
 I installed TrustWallet and MetaMask, login with my Wallet Address, but I openned 2 app and still not see any confirm screen to login. Can you help?
 
 I uploaded the picture here:
 
 
![IMG_4194](https://github.com/user-attachments/assets/c6ca01aa-9a4c-47d8-9104-3da5f27b6c48)

![IMG_4195](https://github.com/user-attachments/assets/eaf9d4fe-6564-4d22-a10e-5c2eb095edc7)

In the app, nothing show:
![IMG_4196](https://github.com/user-attachments/assets/69bbc3d4-8120-45e7-be4d-02b47c644330)

### List of related npm package versions

```
    "@walletconnect/client": "^1.8.0",
    "@web3modal/ethers": "^5.1.0",
    "ethers": "^6.13.2",
    "next": "14.2.5",
    "react": "^18",
    "react-dom": "^18",
```

**Comment by chainxdev** [comment]
Also the same thing happening with the Wagmi version and NextJS

**Comment by chainxdev** [comment]
For me the issue was caused by a wrongly configured walletconnect projectId. Changed it to a new id and it started working.

**Comment by yoheinishikubo** [comment]
I'm encountering the same issue in my setup.

The problem arises when MetaMask unexpectedly switches from the specified `sepolia` network to another network like `mainnet` even after configuring networks as [sepolia] and setting `defaultNetwork` to `sepolia`.

Is this the designed behavior?
I've thoroughly reviewed the documentation and explored potential solutions involving Hooks and Events, but unfortunately, I haven't found a viable workaround.

To reproduce this issue, I've followed the guide at https://docs.walletconnect.com/appkit/next/core/installation, choosing `Etheres` and `Bun` as the basis for my implementation. While I've made some necessary adjustments, the core functionality remains largely unchanged.

**Comment by yoheinishikubo** [comment]
I'm encountering the same issue in my setup.

It appears that this problem always arises when there's a mismatch between the network specified in the code and the network selected in MetaMask.
It works fine when both values are the same.

Is this the designed behavior?
I've thoroughly reviewed the documentation and explored potential solutions involving Hooks and Events, but unfortunately, I haven't found a viable workaround.

To reproduce this issue, I've followed the guide at https://docs.walletconnect.com/appkit/next/core/installation, choosing `Etheres` and `Bun` as the basis for my implementation. While I've made some necessary adjustments, the core functionality remains largely unchanged.
